pymc/gp/hsgp_approx.py

```diff
@@ -44,36 +46,265 @@ def set_boundary(X: TensorLike, c: numbers.Real | TensorLike) -> np.ndarray:
     return L
 
 
-def calc_eigenvalues(L: TensorLike, m: Sequence[int]):
-    """Calculate eigenvalues of the Laplacian."""
+def calc_eigenvalues_dirichlet(L: TensorLike, m: Sequence[int]) -> np.ndarray:
+    """Calculate eigenvalues of the Laplacian for Dirichlet boundary conditions.
+
+    Computes eigenvalues λ_j = (πj/(2L))² for indices j = 1, 2, ..., m.
+
+    Parameters
+    ----------
+    L : array-like
+        The boundary of the approximation domain for each dimension.
+    m : Sequence[int]
+        The number of basis vectors to use for each dimension.
+
+    Returns
+    -------
+    np.ndarray
+        A 2D array of shape ``(prod(m), len(m))`` containing the eigenvalues.
+
+    See Also
+    --------
+    calc_eigenvalues_neumann : Eigenvalues for Neumann boundary conditions.
+    calc_eigenvectors_dirichlet : Corresponding eigenvectors (sine basis).
+
+    Notes
+    -----
+    For Dirichlet boundary conditions, the eigenfunctions satisfy f(±L) = 0
+    (function values are zero at the boundaries).
+
+    References
+    ----------
+    .. [1] Solin, A., Särkkä, S. (2019). Hilbert Space Methods for Reduced-Rank
+           Gaussian Process Regression. Statistics and Computing, 30, 419-446.
+    """
     S = np.meshgrid(*[np.arange(1, 1 + m[d]) for d in range(len(m))])
     S_arr = np.vstack([s.flatten() for s in S]).T
+    return np.square((np.pi * S_arr) / (2 * L))
+
+
+def calc_eigenvalues_neumann(L: TensorLike, m: Sequence[int]) -> np.ndarray:
+    """Calculate eigenvalues of the Laplacian for Neumann boundary conditions.
+
+    Computes eigenvalues λ_j = (πj/(2L))² for indices j = 1, 2, ..., m.
+
+    The constant eigenfunction (j=0) is intentionally excluded to avoid
+    identifiability issues with the model intercept or mean function.
```

Comment:
I find this highly unexpected.

Maybe we should be a separate `mean-zero-neumann` BC?